File: app/src/main/java/com/katdmy/android/unocount/PlayerRepository.java
package com.katdmy.android.unocount;

import android.app.Application;

import androidx.lifecycle.LiveData;

import java.util.List;

class PlayerRepository {
    private PlayerDao mPlayerDao;

    PlayerRepository(Application application) {
        PlayerDatabase db = PlayerDatabase.getDatabase(application);
        mPlayerDao = db.playerDao();
    }

    LiveData<List<Player>> getPlayers() {
        return mPlayerDao.getPlayers();
    }

    LiveData<List<String>> getActivePlayers() {
        return mPlayerDao.getActivePlayers();
    }

    void setActive(String name, boolean active) {
        mPlayerDao.setActive(name, active);
    }

    void addPlayer(Player player) {
        if (mPlayerDao.getPlayerByName(player.getName()).size() == 0)
            mPlayerDao.addPlayer(player);
    }

    void deletePlayer(Player player) {
        mPlayerDao.deletePlayer(player);
    }
}

File: app/src/main/java/com/katdmy/android/unocount/domain/player/PlayerRepository.kt
package com.katdmy.android.unocount.domain.player

import androidx.lifecycle.LiveData
import com.katdmy.android.unocount.domain.player.model.Player

interface PlayerRepository {
    val players: LiveData<List<Player>>
    val activePlayers: LiveData<List<String>>

    fun setActive(playerName: String, active: Boolean)
    fun addPlayer(playerName: String)
    fun deletePlayer(playerName: String)
}